android/runtime/v8/src/native/V8Util.cpp

```diff
@@ -146,17 +144,41 @@ void V8Util::openJSErrorDialog(Isolate* isolate, TryCatch &tryCatch)
 		return;
 	}
 
+	HandleScope scope(isolate);
+
 	Local<Context> context = isolate->GetCurrentContext();
 	Local<Message> message = tryCatch.Message();
-	Local<Value> stack = tryCatch.Exception().As<Object>()->Get(context, STRING_NEW(isolate, "stack")).ToLocalChecked();
+	Local<Value> exception = tryCatch.Exception();
+
+	Local<Value> jsStack;
+	Local<Value> javaStack;
+
+	if (exception->IsObject()) {
+		Local<Object> error = exception.As<Object>();
+		jsStack = exception.As<Object>()->Get(STRING_NEW(isolate, "stack"));
+		javaStack = exception.As<Object>()->Get(STRING_NEW(isolate, "javaStack"));
+	}
 
-	V8Runtime::exceptionStackTrace.Reset(isolate, message->GetStackTrace());
+	// obtain javascript stack trace
+	if (jsStack->IsNullOrUndefined() || jsStack.IsEmpty()) {
+		Local<StackTrace> frames = message->GetStackTrace();
+		if (frames.IsEmpty() || !frames->GetFrameCount()) {
+			frames = StackTrace::CurrentStackTrace(isolate, 10);
+		}
+		if (!frames.IsEmpty()) {
+			std::string stackString = V8Util::stackTraceString(frames);
+			if (!stackString.empty()) {
+				jsStack = String::NewFromUtf8(isolate, stackString.c_str()).As<Value>();
+			}
+		}
+	}
 
 	jstring title = env->NewStringUTF("Runtime Error");
 	jstring errorMessage = TypeConverter::jsValueToJavaString(isolate, env, message->Get());
 	jstring resourceName = TypeConverter::jsValueToJavaString(isolate, env, message->GetScriptResourceName());
 	jstring sourceLine = TypeConverter::jsValueToJavaString(isolate, env, message->GetSourceLine());
-	jstring javaStack = TypeConverter::jsValueToJavaString(isolate, env, stack);
+	jstring jsStackString = TypeConverter::jsValueToJavaString(isolate, env, jsStack);
```

Comment:
I think you have to guard here for empty handle and/or undefined. TypeConverter handles `null` -> `NULL`, but I think it converts `undefined` to `'undefined'`. And I think it'll just crash if the handle is empty.

I assigned to a bool up above for when the stack/nativeStack were not null/undefined/empty and if they were, I assigned NULL here and avoided the `DeleteLocalRef` below.

Reply:
I think we should improve `TypeConverter` to handle this validation?

Updated PR